### DISCUSSION TITLE
Stylix + Nixvim/Neovim - Light themes always have a grey background

### DISCUSSION THREAD

**Discussion by dmarcoux** [nix-community/stylix#590] [Q&A] [answered]
Hi,

I'm setting up Stylix to style Neovim (configured via Nixvim) and for an unknown reason, Neovim always has a grey background whenever I set Stylix to use a light theme such as `solarized-light`.

Here's an example of how this looks in Neovim:
![example](https://github.com/user-attachments/assets/c05b18a6-2929-4eb0-8af4-1798f43074de)

With `solarized-light`, the background should be a light beige, not grey. In my Neovim config, I have `vim.opt.termguicolors = true` and my terminal emulator is Alacritty. The only way to have the theme correctly render in Neovim is to disable the styling of Alacritty with `stylix.targets.alacritty.enable = false;`.

Did someone encounter this? How can I fix this issue?

**Comment by trueNAHO** [comment, marked as answer]
> I'm setting up Stylix to style Neovim (configured via Nixvim) and for an unknown reason, Neovim always has a grey background whenever I set Stylix to use a light theme such as `solarized-light`.
>
> [...]
>
> With `solarized-light`, the background should be a light beige, not grey. In my Neovim config, I have `vim.opt.termguicolors = true` and my terminal emulator is Alacritty. The only way to have the theme correctly render in Neovim is to disable the styling of Alacritty with `stylix.targets.alacritty.enable = false;`.

Maybe the `stylix.targets.nixvim.transparentBackground` options resolve your issue:

https://github.com/danth/stylix/blob/5699ba97c60455ebafde0fd4e78ca0a2e5a58282/modules/nixvim/nixvim.nix#L10-L13

Currently, Neovim and Nixvim are styled through the `base16` plugin:

https://github.com/danth/stylix/blob/5699ba97c60455ebafde0fd4e78ca0a2e5a58282/modules/nixvim/nixvim.nix#L44-L52

Alternatively, take a look at https://github.com/danth/stylix/pull/536.

**Reply by dmarcoux** [reply]
I tried enabling and disabling `stylix.targets.nixvim.transparentBackground`, it didn't change anything in both cases.

Beside this, I applied the changes from the PR you linked (#536) and it works! The background is now using the expected color for my theme.
